### ISSUE TITLE
404 when using GridFieldBulkUpload

### ISSUE THREAD

**Issue by hailwood** [unclecheese/silverstripe-kickassets#18]
Hey Cheese,

So I've been bugging you enough on IRC I thought I'd actually work on it myself.

Currently the fix is to apply this extension to GridFieldBulkUpload_Request.

Sounds like it's actually an issue with the GridFieldBulkUpload not passing requests onto the uploadfield if it doesn't understand them so maybe I should log a bug there... Either Way.

````
<?php

class GridFieldBulkUpload_RequestExtension extends Extension {

    private static $allowed_actions = [
        'folderid'
    ];

    public function getdisplayfoldername() {
        /** @var UploadField $uploadField */
        $uploadField = $this->owner->getUploadField();

        return $uploadField->getFolderName();
    }

    public function folderid() {
        /** @var UploadField $uploadField */
        $uploadField = $this->owner->getUploadField();

        return $uploadField->folderid(Controller::curr()->getRequest());
    }
}
````

**Comment by unclecheese** [comment]
It looks to me like a malformed URL from your bulk upload module. My guess is that the "select existing file" button doesn't work, either, because it uses the same method to generate that URL.

**Comment by hailwood** [comment]
@unclecheese Correct, the getdisplayfoldername is actually fixing an issue from another module with a similar issue when used with GridFieldBulkUpload. It looks like the GridFieldBulkUpload intercepts requests that would normally hit the upload field directly. But all the other modules expect it to be hitting the UploadField.

**Comment by unclecheese** [comment]
Probably easier to just use a Dropzone field. :) UploadField is sooooo 2012.

**Comment by hailwood** [comment]
Agreed, I like Dropzonefield, I'm gonna look into it so I'll probably bug you on IRC about that too ;) but until GridFieldBulkUpload fix things this extension should be useful for anyone with the same issue!
